### PR TITLE
QT: add progress plan Graph

### PR DESCRIPTION
**Pull request by shishovKA**
https://nda.ya.ru/t/WnmZs6Kl6vUrHD

how to check this huge PR:

1st commit is the biggest one, but it contains only "copy-paste" files for yql-like graph

other commits are more informative, they are contain:
 - installing libs
 - hiding JS webworker functionality (for now decided not to use worker cause build lib troubles)
 - reworking graph nodes onClick handlers (open YT table or YT operation)
 - replacing some internal component with ours analogs

### CODE REVIEW

File: packages/ui/src/ui/pages/query-tracker/Plan/models/yql/yql.ts [hunk 1]
```diff
@@ -197,7 +197,7 @@ export interface definitions {
     NodeProgress: {
         category?: string;
         remoteId?: string;
-        state?: 'STARTED' | 'IN_PROGRESS' | 'FINISHED' | 'FAILED' | 'ABORTED';
+        state?: 'Started' | 'InProgress' | 'Finished' | 'Failed' | 'Aborted';
```
[the remark this answers is not in the document]
Comment (shishovKA) [reply packages/ui/src/ui/pages/query-tracker/Plan/models/yql/yql.ts:200]: These values come from backend. QT team has noted that they are going to be CamelCase

File: packages/ui/src/ui/pages/query-tracker/Plan/utils.ts [hunk 2]
```diff
@@ -618,3 +623,28 @@ export function getConnectedEdges(
...
+            } else if (node.progress?.remoteId) {
+                node.url = getOperationUrl(node, buildOperationUrlLocal);
+            }
+
+            return node;
+        },
+        [buildOperationUrlLocal],
+    );
+}
+*/
```
[the remark this answers is not in the document]
Comment (shishovKA) [reply packages/ui/src/ui/pages/query-tracker/Plan/utils.ts]: this function probably was reworked and uncommented in next commits

File: packages/ui/src/ui/pages/query-tracker/Plan/services/getWorkers.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
-//import {Worker} from '@gravity-ui/app-builder/worker';
+/*
+import {Worker} from '@gravity-ui/app-builder/worker';
 
 export function getBasicLayoutWorker() {
-    console.log(new URL('./layout.worker.ts'));
-    return new Worker(new URL('./layout.worker.ts'));
+    return new Worker(new URL('./layout.worker.ts', import.meta.url));
 }
+*/
```
[the remark this answers is not in the document]
Comment (shishovKA) [reply packages/ui/src/ui/pages/query-tracker/Plan/services/getWorkers.ts:7]: i will delete this file 👍🏻

File: packages/ui/src/ui/pages/query-tracker/Plan/Graph.tsx [hunk 4]
```diff
@@ -6,11 +6,9 @@ import Loader from './components/Loader/Loader';
...
 import {useUpdate} from 'react-use';
 import ResizeObserver from 'resize-observer-polyfill';
-import {getBasicLayoutWorker} from './services/getWorkers';
-import type {LayoutWorkerFunctions} from './services/layout.worker';
-// import {countEvent} from 'services/metrics';
-import cn from 'bem-cn-lite';
-import {PromisifiedWorker, promisifyWorker} from './utils/promisified-worker';
+/* import {getBasicLayoutWorker} from './services/getWorkers'; */
+// import type {LayoutWorkerFunctions} from './services/layout.worker';
+// import {PromisifiedWorker /* promisifyWorker */} from './utils/promisified-worker';
```
[the remark this answers is not in the document]
Comment (shishovKA) [reply packages/ui/src/ui/pages/query-tracker/Plan/Graph.tsx]: I've removed these comments 👍🏻

File: packages/ui/src/ui/pages/query-tracker/Plan/NodeSchemas.scss [hunk 5]
```diff
@@ -1,5 +1,5 @@
 .yql-node-schemas {
-    &__item + &__item {
+    &__item.schema + &__item.schema {
```
[the remark this answers is not in the document]
Comment (shishovKA) [reply packages/ui/src/ui/pages/query-tracker/Plan/NodeSchemas.scss]: I've reworked these classes as: `&__item_schema`, `&__item_title`

File: packages/ui/src/ui/pages/query-tracker/Plan/NodeSchemas.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+interface SchemasSectionProps {
+    title: string;
+    schemas: {name: string; type: yqlModel.value.TypeArray}[];
+}
+function SchemasSection({title, schemas}: SchemasSectionProps) {
+    return (
+        <Collapse
+            className={[block('item'), block('item_title')].join(' ')}
```
Comment (ma-efremoff) [inline packages/ui/src/ui/pages/query-tracker/Plan/NodeSchemas.tsx:38]: Regular way to add modifiers is:
```ts
className={block('item', {title: true})}
```

File: packages/ui/src/ui/pages/query-tracker/Plan/NodeSchemas.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                    const schemaFields =
+                        result.resultType === 'table' || result.resultType === 'row'
+                            ? result.scheme
+                            : undefined;
+                    if (!schemaFields) {
+                        return null;
+                    }
+                    return (
+                        <Collapse
+                            className={[block('item'), block('item_schema')].join(' ')}
```
Comment (ma-efremoff) [inline packages/ui/src/ui/pages/query-tracker/Plan/NodeSchemas.tsx:55]: the same

**Review by ma-efremoff — APPROVED** [review]

**Pull request merged**